gen/golang/templates/client.go.tmpl

```diff
@@ -21,20 +23,30 @@ const {{.Name | constPathPrefix}} = "/rpc/{{.Name}}/"
 {{range .Services}}
   {{ $serviceName := .Name | clientServiceName}}
   type {{$serviceName}} struct {
-    client HTTPClient
-    urls   [{{.Methods | countMethods}}]string
+    client         HTTPClient
+    urls           [{{.Methods | countMethods}}]string
+    InjectFunction InjectHTTPRequestFunction
```

Comment:
make it a pointer then you don't have to `...` and `[0]`?

Reply:
do you want it to be a mandatory parameter instead of optional?